piece.cc

```diff
@@ -52,10 +53,80 @@ int Piece::move(Piece *currentTile, Piece *targetTile, int newX, int newY) {
             std::swap(currentTile, targetTile);
         }
 
+        // check if move was a King castling move
+        if (targetTile->getName() == "K" && (*legalMovesMap)[targetBox] == 2) {
+            // to find which rook we are moving towards
+            if (yCoord - newY > 0) {
+                for (auto &pair: (*((*board)[7][0]->getLegalMoves()))) {
+                    if (pair.second == 2) {
+                        std::swap((*board)[7][0], (*board)[pair.first.getX()][pair.first.getY()]);
+                        (*board)[pair.first.getX()][pair.first.getY()]->xCoord = 7;
+                        (*board)[pair.first.getX()][pair.first.getY()]->yCoord = 0;
+                        (*board)[7][0]->xCoord = pair.first.getX();
+                        (*board)[7][0]->yCoord = pair.first.getY();
```

Comment:
B/c the `(*board)[pair.first.getX()][pair.first.getY()]` is now pointing at the rook AFTER THE SWAP, shouldn't the `xCoord` and `yCoord` be just updated to `pair.first.getX()` and `pair.first.getY()`?

Also, B/c the rook is moving into a nullptr, taking `(*board)[7][0]->xCoord` will result in an error b/c `*board)[7][0]` is null. Hence, you don't even need to change any xCoords or yCoords for nullptrs on the board.

Reply:
you are totally right